cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/RepositoryManagerTest.java

```diff
@@ -0,0 +1,172 @@
+                                                            .build();
+    SourceControlConfig sourceControlConfig = new SourceControlConfig(new NamespaceId(NAMESPACE), config, cConf);
+    boolean exceptionCaught = false;
+    try {
+      RepositoryManager.validateConfig(secureStore, sourceControlConfig);
+    } catch (RepositoryConfigValidationException e) {
+      exceptionCaught = true;
+      Assert.assertTrue(e.getMessage().contains("Failed to get auth credentials"));
+    }
+    Assert.assertTrue(exceptionCaught);
+    testRepositoryDirectoryDeleted();
+  }
+
+  @Test
+  public void testValidateInvalidToken() throws Exception {
+    String serverURL = gitServer.getServerURL();
+    RepositoryConfig config = new RepositoryConfig.Builder().setProvider(Provider.GITHUB)
+                                                            .setLink(serverURL + "ignored")
+                                                            .setDefaultBranch("develop")
+                                                            .setAuthType(AuthType.PAT)
+                                                            .setTokenName(GITHUB_TOKEN_NAME)
+                                                            .build();
+    SourceControlConfig sourceControlConfig = new SourceControlConfig(new NamespaceId(NAMESPACE), config, cConf);
+    Mockito.when(secureStore.get(NAMESPACE, GITHUB_TOKEN_NAME))
+           .thenReturn(new SecureStoreData(null, "invalid-token".getBytes(StandardCharsets.UTF_8)));
+
+    boolean exceptionCaught = false;
+    try {
+      RepositoryManager.validateConfig(secureStore, sourceControlConfig);
+    } catch (RepositoryConfigValidationException e) {
+      exceptionCaught = true;
+      Assert.assertTrue(e.getMessage().contains("Authentication credentials invalid"));
+    }
+    Assert.assertTrue(exceptionCaught);
+    testRepositoryDirectoryDeleted();
+  }
+
+  @Test
+  public void testValidateInvalidBranchName() throws Exception {
+    String serverURL = gitServer.getServerURL();
+    RepositoryConfig config = new RepositoryConfig.Builder().setProvider(Provider.GITHUB)
+                                                            .setLink(serverURL + "ignored")
+                                                            .setDefaultBranch("develop-invalid")
+                                                            .setAuthType(AuthType.PAT)
+                                                            .setTokenName(GITHUB_TOKEN_NAME)
+                                                            .build();
+    SourceControlConfig sourceControlConfig = new SourceControlConfig(new NamespaceId(NAMESPACE), config, cConf);
+    boolean exceptionCaught = false;
+    try {
+      RepositoryManager.validateConfig(secureStore, sourceControlConfig);
+    } catch (RepositoryConfigValidationException e) {
+      exceptionCaught = true;
+      System.out.println("Arjan look here " + e.getMessage());
```

Comment:
Please remove system.out

Reply:
That was used for debugging, should not have been pushed. Removed it.